smarts/core/utils/traffic_history_serivce.py

```diff
@@ -77,6 +80,17 @@ def __init__(self, history_file_path):
         self._prepare_next_batch()
         self._receive_data_conn.recv()
 
+    def teardown(self):
+        if self.is_in_use:
```

Comment:
I don't see where `history_file_path` gets reset to `None`.  So would this `is_in_use` check always succeed?  If so, then this `teardown()` method isn't really idempotent.

Reply:
Right, nice catch. thanks